front-end/src/screens/Settings/address.tsx

```diff
@@ -183,74 +175,117 @@ const Address = ({ className }: Props): JSX.Element => {
 		);
 	}
 
-	if (accounts.length === 0) {
-		return (
-			<Form className={className} standalone={false}>
-				<Form.Group>
-					<Form.Field width={16}>
-						<label>Linked address</label>
-						<div className='text-muted'>Associate your account with an on chain address using the <a href={getExtensionUrl()}>Polkadot-js extension</a>.</div>
-						<div className='link-button-container'>
-							<Button primary onClick={handleDetect}>
-								{currentUser?.addresses?.length ? 'Show linked addresses' : 'Link address'}
-							</Button>
-						</div>
-					</Form.Field>
-				</Form.Group>
-			</Form>
-		);
-	}
-
 	return (
 		<Form className={className} standalone={false}>
-			<Form.Group>
-				<Form.Field width={16}>
-					<label className='header'>Available addresses</label>
-					<div className='ui list'>
-						{accounts.map(account => {
-							const address = getEncodedAddress(account.address);
+			{accounts.length === 0 ?
+				<>
+					{currentUser?.addresses?.length ? <Form.Group>
+						<Form.Field width={16}>
+							<label className='header'>Linked addresses</label>
+							<div className='ui list'>
+								{currentUser?.addresses?.sort().map(userAddress => {
+									const address = getEncodedAddress(userAddress);
+
+									return address &&
+										<Grid key={address}>
+											<Grid.Column width={7}>
+												<div className='item'>
+													<AddressComponent className='item' address={address} />
+												</div>
+											</Grid.Column>
+											<Grid.Column width={3}>
+												<div className='button-container'>
+													<Button
+														className={'social'}
+														negative={true}
+														onClick={() => handleUnlink(address)}
+													>
+														{unlinkIcon}
+													</Button>
+												</div>
+											</Grid.Column>
+											<Grid.Column width={6} >
+												{currentUser.defaultAddress !== address ?
```

Comment:
Those 2 conditions `currentUser.defaultAddress !== address` and `currentUser.defaultAddress === address` should be combined. Also the same code is used a couple lines below, this should be extracted in a function returning this component.